fvm/meter/interaction_meter.go

```diff
@@ -51,32 +57,43 @@ func NewInteractionMeter(params InteractionMeterParameters) *InteractionMeter {
 func (m *InteractionMeter) MeterStorageRead(
 	storageKey StorageInteractionKey,
 	value flow.RegisterValue,
-	enforceLimit bool) error {
+	enforceLimit bool,
+) error {
 
 	// all reads are on a View which only read from storage at the first read of a given key
-	if _, ok := m.storageUpdateSizeMap[storageKey]; !ok {
+	if _, ok := m.reads[storageKey]; !ok {
```

Comment:
if you unconditionally set a value (without reading it first), then get it, you will be now count that against both total bytes read and total byte written.  this is more aggressive than the original implementation, which only counts against bytes written in this scenario.

seems like what you really want is to ensure the key is in neither reads / writes sets here, and when you merge, "the sum of the union of child's reads set with parent's writes set" should be discounted (child read entries should not be included in the parent's reads set if the entry is in the parent's write set)